xcdat/dataset.py

```diff
@@ -35,13 +37,97 @@
 ]
 
 
-def open_dataset(
+def open(
     path: str,
     data_var: Optional[str] = None,
     add_bounds: bool = True,
     decode_times: bool = True,
     center_times: bool = False,
     lon_orient: Optional[Tuple[float, float]] = None,
+    **kwargs,
+) -> xr.Dataset:
+    """
+    Documentation to be written when this PR is
+    further along...
+    """
+
+    # inspect path attributes and use either open_dataset
+    # or open_mfdataset as appropriate
+    #   * list (open list of files with open_mfdataset)
+    #   * pathlib.Path (open pathlib.Path with open_mfdataset)
+    #   * deprecated CDAT xml file type
+    #   * directory (opens with open_dataset or open_mfdataset
+    #                depending in number of files)
+    #   * wildcard input (opens data with open_mfdataset)
```

Comment:
From my understanding, the motivation behind adding a new `open` function is to provide an abstracted API that determines whether to use `open_dataset()`/`open_mfdataset()` based on the `path` argument to make opening datasets hopefully more intuitive. Additionally, `open` would include support for the deprecated CDAT XML format.

Let me know if this _isn't_ correct.